test/unit/config.test.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const os = require('os');
+const mockFs = require('mock-fs');
+
+const {
+  getHomeDir,
+  getNcuDir,
+  getConfigPath,
+  getConfig,
+  getMergedConfig,
+  writeConfig,
+  updateConfig
+} = require('../../lib/config');
+
+const username = 'foo';
+const token = '123456789';
+const globalFileContent =
+  `{
+    "username": "${username}",
+    "token": "${token}"
+  }`;
+const localUsername = 'local';
+const localFileContent =
+  `{
+    "username": "${localUsername}"
+  }`;
+const newUsername = 'bar';
+const newToken = 'asdfghj';
+
+describe('config', () => {
+  beforeEach(() => {
+    mockFs({
+      // Mock global .ncurc
+      [path.join(os.homedir(), '.ncurc')]: globalFileContent,
```

Comment:
If the user has `XDG_CONFIG_HOME` set when running the tests, this is not enough right? We need to manually set `XDG_CONFIG_HOME` to undefined in the before hook to make sure it does not read from there.

Reply:
Yes, you are right. Thanks @joyeecheung!